library/geocore/src/main/java/it/geosolutions/imageio/compression/Decompressor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *    ImageI/O-Ext - OpenSource Java Image translation Library
+ *    https://www.geosolutionsgroup.com/
+ *    https://github.com/geosolutions-it/imageio-ext
+ *    (C) 2022, GeoSolutions
+ *    All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of GeoSolutions nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY GeoSolutions ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GeoSolutions BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package it.geosolutions.imageio.compression;
+
+import java.util.zip.DataFormatException;
+
+/**
+ * Interface for byte array Decompressor.
+ *
+ * Decompressor should be first initialized calling setInput on a source data buffer to be decompressed.
+ * Afterwards, the decompress method should be called to decompress the provided source into
+ * destData buffer.
+ */
+public interface Decompressor {
+
+    /** Set the input source data buffer to be decompressed */
+    void setInput(byte [] srcData);
+
+    /**
+     * Decompress the source data and put result in the dest data buffer, starting from offset
+     * position, with a maximum number of uncompressed bytes
+     */
+    void decompress(byte[] destData, int offset, int maxUncompressedSize) throws DataFormatException;
```

Comment:
Same note about srcData vs destData as in compressor, plus noting that the method signatures are not similar (again, thinking out loud, one might think they should be similar in design, but maybe there is a reason for them to differ).

Reply:
Both implementations of the decompressors simply decompress whatever has been provided as source.